### PR TITLE
Removed the scales from the Plot constructors

### LINKED ISSUE #1507: Remove scales from Plot constructors
The presence of `Scale`s as arguments to `Plot` constructors seems redundant. Consider:

```Javascript
var plot = new Plottable.Plot.Scatter(xScale, yScale);
...
plot.project("x", ..., xScale);
plot.project("y", ..., yScale);
```

The user already supplies the `Scale`s in the `project()` call, so it seems like we shouldn't need to specify them in the constructor.

### PR DESCRIPTION
**Pull request by aicioara**
Closes #1507 

They are specified in the `.x()` , `.y()` bindings, so they were redundant.

The `try` `catch` blocks in the quicktests will be removed after passing QE

**Alternative implementation considered**: specify the scales in the constructor and do not specify them in the projectors (`.x()`, `.y()`, etc). Might not be good because then we can't change the scale for a plot

**Update guide**
Refactor RegEx for the update
```regex
(\.Plots\.[^\(]*\()[^,\)]*(, )?[^\),]*(, )*
Replace with
$1
```

**QE notes**:
This change affected all the plots and their interaction with the scales. The old behavior should not be changed in any way. In particular if something went wrong in this PR, the axes (because they rely on scales) should behave weirdly or the data should be all over the place. Edge cases I could think of and manually tested so far included
* adding datasets before calling `.x()`, or `.y()`, 
* calling `.y0()` before `.y()` on `AreaPlot`
* changing scales half way through and update them only on one of the projectors (only on `y0` for `AreaPlot` or only `x2` on `RectanglePlot`, etc) 
* Shifting initialization statements around 

**Release Notes**
## Plots now no longer require Scales to be passed in the constructor
Old
```typescript
var plot = new Plottable.Plots.Bar(xScale, yScale);
``` 
New
```typescript
var plot = new Plottable.Plots.Bar();
```

The new signatures should now be 
* `new Plottable.Plot()`
* `new Plottable.XYPlot<type, type>()`
* `new Plottable.Plots.Pie()` <- unchanged
* `new Plottable.Plots.Bar<type, type>(orientation?)`
* `new Plottable.Plots.StackedBar<type, type>(orientation?)`
* `new Plottable.Plots.ClusteredBar<type, type>(orientation?)`
* `new Plottable.Plots.Line<type>()`
* `new Plottable.Plots.Area<type>()`
* `new Plottable.Plots.StackedArea<type>()`
* `new Plottable.Plots.Rectangle<type, type>()`
* `new Plottable.Plots.Scatter<type, type>()`

The scales are now deduced from the `.x()`, `.y()`, etc calls. Major implication is that plots can no longer be rendered on the svg unless the projector methods (`.x()`, `.y()`) have been called beforehand (which makes sense)

Upgrade Regex
```regex
(\.Plots\.[^\(]*\()[^,\)]*(, )?[^\),]*(, )*
Replace with
$1
```

### CODE REVIEW

File: src/components/plots/areaPlot.ts [hunk 1]
```diff
@@ -15,10 +15,10 @@ export module Plots {
      * @param {QuantitativeScale} xScale The x scale to use.
      * @param {QuantitativeScale} yScale The y scale to use.
      */
-    constructor(xScale: QuantitativeScale<X>, yScale: QuantitativeScale<number>) {
-      super(xScale, yScale);
+    constructor() {
+      super();
       this.classed("area-plot", true);
-      this.y0(0, yScale); // default
+      this.y0(); // default
```
Comment (bluong) [inline src/components/plots/areaPlot.ts]: Should this not be `this.y0(0)`?

File: src/components/plots/areaPlot.ts [hunk 2]
```diff
@@ -26,13 +26,32 @@ export module Plots {
...
     }
 
+    public y(): Plots.AccessorScaleBinding<number, number>;
+    public y(y: number | Accessor<number>): Area<X>;
+    public y(y: number | Accessor<number>, yScale: QuantitativeScale<number>): Area<X>;
+    public y(y?: number | Accessor<number>, yScale?: QuantitativeScale<number>): any {
+      if (y == null) {
+        return super.y();
+      }
+      super.y(y, yScale);
```
Comment (bluong) [inline src/components/plots/areaPlot.ts]: To keep with consistency, you probably want to do
```typescript
if (yScale == null) {
  super.y(y);
} else {
  super.y(y, yScale);
}
```

File: src/components/plots/areaPlot.ts [hunk 3]
```diff
@@ -26,13 +26,32 @@ export module Plots {
...
+
+      if (yScale != null) {
+        var y0Binding = this.y0();
+        var y0 = y0Binding && y0Binding.accessor;
```
Comment (bluong) [inline src/components/plots/areaPlot.ts]: y0 is always predefined to be at least to be 0 so y0 can be assumed to be nonnull i believe

File: src/components/plots/areaPlot.ts [hunk 4]
```diff
@@ -26,13 +26,32 @@ export module Plots {
       this.attr("stroke", defaultColor);
     }
 
+    public y(): Plots.AccessorScaleBinding<number, number>;
+    public y(y: number | Accessor<number>): Area<X>;
```
Comment (bluong) [inline src/components/plots/areaPlot.ts]: The above declarations are unnecessary

File: src/components/plots/areaPlot.ts [hunk 5]
```diff
@@ -26,13 +26,32 @@ export module Plots {
...
     public y0(): Plots.AccessorScaleBinding<number, number>;
     public y0(y0: number | Accessor<number>): Area<X>;
-    public y0(y0: number | Accessor<number>, y0Scale: Scale<number, number>): Area<X>;
-    public y0(y0?: number | Accessor<number>, y0Scale?: Scale<number, number>): any {
+    public y0(y0?: number | Accessor<number>): any {
       if (y0 == null) {
         return this._propertyBindings.get(Area._Y0_KEY);
       }
+      var y0Binding = this.y();
+      var y0Scale = y0Binding && y0Binding.scale;
```
Comment (bluong) [inline src/components/plots/areaPlot.ts]: `yBinding`?

File: src/components/plots/barPlot.ts [hunk 6]
```diff
@@ -38,6 +38,30 @@ export module Plots {
       this.attr("width", () => this._getBarPixelWidth());
     }
 
+    public x(): Plots.AccessorScaleBinding<X, number>;
+    public x(x: number | Accessor<number>): Bar<X, Y>;
+    public x(x: X | Accessor<X>, xScale: Scale<X, number>): Bar<X, Y>;
```
Comment (bluong) [inline src/components/plots/barPlot.ts]: Above declarations are unnecessary

File: src/components/plots/barPlot.ts [hunk 7]
```diff
@@ -38,6 +38,30 @@ export module Plots {
...
+        return super.x();
+      }
+      super.x(<X | Accessor<X>>x, xScale);
+      this._updateValueScale();
+      return this;
+    }
+
+    public y(): Plots.AccessorScaleBinding<Y, number>;
+    public y(y: number | Accessor<number>): Bar<X, Y>;
+    public y(y: Y | Accessor<Y>, yScale: Scale<Y, number>): Bar<X, Y>;
```
Comment (bluong) [inline src/components/plots/barPlot.ts]: Aboe decalarations are unnecessary

File: src/components/plots/barPlot.ts [hunk 8]
```diff
@@ -38,6 +38,30 @@ export module Plots {
...
     }
 
+    public x(): Plots.AccessorScaleBinding<X, number>;
+    public x(x: number | Accessor<number>): Bar<X, Y>;
+    public x(x: X | Accessor<X>, xScale: Scale<X, number>): Bar<X, Y>;
+    public x(x?: number | Accessor<number> | X | Accessor<X>, xScale?: Scale<X, number>): any {
+      if (x == null) {
+        return super.x();
+      }
+      super.x(<X | Accessor<X>>x, xScale);
```
Comment (bluong) [inline src/components/plots/barPlot.ts]: Split this up into the appropriate 2 setters please

File: src/components/plots/barPlot.ts [hunk 9]
```diff
@@ -38,6 +38,30 @@ export module Plots {
...
+    }
+
+    public y(): Plots.AccessorScaleBinding<Y, number>;
+    public y(y: number | Accessor<number>): Bar<X, Y>;
+    public y(y: Y | Accessor<Y>, yScale: Scale<Y, number>): Bar<X, Y>;
+    public y(y?: number | Accessor<number> | Y | Accessor<Y>, yScale?: Scale<Y, number>): any {
+      if (y == null) {
+        return super.y();
+      }
+      super.y(<Y | Accessor<Y>>y, yScale);
```
Comment (bluong) [inline src/components/plots/barPlot.ts]: Split this up into the appropriate 2 setters please

File: src/components/plots/barPlot.ts [hunk 10]
```diff
@@ -38,6 +38,30 @@ export module Plots {
...
 
+    public x(): Plots.AccessorScaleBinding<X, number>;
+    public x(x: number | Accessor<number>): Bar<X, Y>;
+    public x(x: X | Accessor<X>, xScale: Scale<X, number>): Bar<X, Y>;
+    public x(x?: number | Accessor<number> | X | Accessor<X>, xScale?: Scale<X, number>): any {
+      if (x == null) {
+        return super.x();
+      }
+      super.x(<X | Accessor<X>>x, xScale);
+      this._updateValueScale();
```
Comment (bluong) [inline src/components/plots/barPlot.ts]: Why is this added?

File: src/components/plots/barPlot.ts [hunk 11]
```diff
@@ -38,6 +38,30 @@ export module Plots {
...
+
+    public y(): Plots.AccessorScaleBinding<Y, number>;
+    public y(y: number | Accessor<number>): Bar<X, Y>;
+    public y(y: Y | Accessor<Y>, yScale: Scale<Y, number>): Bar<X, Y>;
+    public y(y?: number | Accessor<number> | Y | Accessor<Y>, yScale?: Scale<Y, number>): any {
+      if (y == null) {
+        return super.y();
+      }
+      super.y(<Y | Accessor<Y>>y, yScale);
+      this._updateValueScale();
```
Comment (bluong) [inline src/components/plots/barPlot.ts]: Why is this added?

File: src/components/plots/rectanglePlot.ts [hunk 12]
```diff
@@ -82,23 +74,32 @@ export module Plots {
     public x(x: number | Accessor<number>): Plots.Rectangle<X, Y>;
     public x(x: X | Accessor<X>, scale: Scale<X, number>): Plots.Rectangle<X, Y>;
     public x(x?: number | Accessor<number> | X | Accessor<X>, scale?: Scale<X, number>): any {
-      if (x == null) {
-        return super.x();
+
+      if (scale != null) {
+        var x2Binding = this.x2();
+        var x2 = x2Binding && x2Binding.accessor;
+        this._bindProperty(Rectangle._X2_KEY, x2, scale);
```
Comment (bluong) [inline src/components/plots/rectanglePlot.ts]: @rcchen can you verify that it makes sense for x2 to never be set with a different scale than x?

File: src/components/plots/rectanglePlot.ts [hunk 13]
```diff
@@ -82,23 +74,32 @@ export module Plots {
...
-    public x2(x2?: number | Accessor<number> | X | Accessor<X>, scale?: Scale<X, number>): any {
+    public x2(x2: number | Accessor<number> | X | Accessor<X>): Plots.Rectangle<X, Y>;
+    public x2(x2?: number | Accessor<number> | X | Accessor<X>): any {
       if (x2 == null) {
         return this._propertyBindings.get(Rectangle._X2_KEY);
       }
-      this._bindProperty(Rectangle._X2_KEY, x2, scale);
+
+      var x2Binding = this.x();
+      var x2Scale = x2Binding && x2Binding.scale;
```
Comment (bluong) [inline src/components/plots/rectanglePlot.ts]: `xBinding` / `xScale`?

File: src/components/plots/rectanglePlot.ts [hunk 14]
```diff
@@ -107,23 +108,32 @@ export module Plots {
...
+    public y2(y2: number | Accessor<number> | Y | Accessor<Y>): Plots.Rectangle<X, Y>;
+    public y2(y2?: number | Accessor<number> | Y | Accessor<Y>): any {
       if (y2 == null) {
         return this._propertyBindings.get(Rectangle._Y2_KEY);
       }
-      this._bindProperty(Rectangle._Y2_KEY, y2, scale);
+
+      var y2Binding = this.y();
+      var y2Scale = y2Binding && y2Binding.scale;
+      this._bindProperty(Rectangle._Y2_KEY, y2, y2Scale);
```
Comment (bluong) [inline src/components/plots/rectanglePlot.ts]: `yBinding` /  `yScale`

File: src/components/plots/stackedAreaPlot.ts [hunk 15]
```diff
@@ -37,7 +37,7 @@ export module Plots {
       this._baseline = this._renderArea.append("line").classed("baseline", true);
     }
 
-    public x(x?: number | Accessor<number> | X | Accessor<X>, xScale?: Scale<X, number>): any {
+    public x(x?: number | Accessor<number> | X | Accessor<X>, xScale?: QuantitativeScale<X>): any {
```
Comment (bluong) [inline src/components/plots/stackedAreaPlot.ts]: Probably should redeclare the appropriate the appropriate setter

File: src/components/plots/stackedAreaPlot.ts [hunk 16]
```diff
@@ -52,7 +52,7 @@ export module Plots {
       return this;
     }
 
-    public y(y?: number | Accessor<number>, yScale?: Scale<number, number>): any {
+    public y(y?: number | Accessor<number>, yScale?: QuantitativeScale<number>): any {
```
Comment (bluong) [inline src/components/plots/stackedAreaPlot.ts]: Probably should redeclare the appropriate the appropriate setter

File: src/components/plots/xyPlot.ts [hunk 17]
```diff
@@ -48,6 +39,12 @@ module Plottable {
       if (this._autoAdjustYScaleDomain) {
         this._updateYExtentsAndAutodomain();
       }
+
+      // TODO: is this needed?
+      if (xScale != null) {
+        xScale.onUpdate(this._adjustYDomainOnChangeFromXCallback);
+      }
```
Comment (bluong) [inline src/components/plots/xyPlot.ts]: Let's fix the TODO

File: src/components/plots/xyPlot.ts [hunk 18]
```diff
@@ -64,6 +61,12 @@ module Plottable {
       if (this._autoAdjustXScaleDomain) {
         this._updateXExtentsAndAutodomain();
       }
+
+      // TODO: is this needed?
+      if (yScale != null) {
+        yScale.onUpdate(this._adjustXDomainOnChangeFromYCallback);
+      }
```
Comment (bluong) [inline src/components/plots/xyPlot.ts]: Let's fix the TODO

File: src/components/plots/xyPlot.ts [hunk 19]
```diff
@@ -225,7 +230,12 @@ module Plottable {
     }
 
     protected _projectorsReady() {
-      return this.x().accessor != null && this.y().accessor != null;
+      var xBinding = this.x();
+      var yBinding = this.y();
+      if (xBinding == null || yBinding == null) {
+        return false;
+      }
+      return xBinding.accessor != null && yBinding.accessor != null;
```
Comment (bluong) [inline src/components/plots/xyPlot.ts]: You can collapse the last two boolean statements

File: src/components/plots/areaPlot.ts (hunk 2 above)
Comment (aicioara) [inline src/components/plots/areaPlot.ts]: I modified this everywhere, that's a bit too verbose and has same effect

File: src/components/plots/areaPlot.ts (hunk 4 above)
Comment (aicioara) [inline src/components/plots/areaPlot.ts]: What do you mean?

File: src/components/plots/barPlot.ts (hunk 6 above)
Comment (aicioara) [inline src/components/plots/barPlot.ts]: What do you mean?

File: src/components/plots/barPlot.ts (hunk 10 above)
Comment (aicioara) [inline src/components/plots/barPlot.ts]: For `barPlot`, the constructor is calling `this.baseline(0)`, which calls `this._updateValueScale`, but there is no scale, so this needs to be called here as a workaround. Also, we don't know if `x` or `y` is the value scale, so has to be called in 2 places

File: src/components/plots/barPlot.ts (hunk 11 above)
Comment (aicioara) [inline src/components/plots/barPlot.ts]: same as above

File: src/components/plots/xyPlot.ts (hunk 17 above)
Comment (aicioara) [inline src/components/plots/xyPlot.ts]: It's a question, left if there to remember.
@jtlan Is this needed? It used to be in the constructor, but not in the projector, which makes me thing it was a bug

File: src/components/plots/stackedAreaPlot.ts (hunk 15 above)
Comment (aicioara) [inline src/components/plots/stackedAreaPlot.ts]: what do you mean?

File: src/components/plots/xyPlot.ts (hunk 19 above)
Comment (aicioara) [inline src/components/plots/xyPlot.ts]: what do you mean?

**Comment by aicioara** [comment]
fixing the overlaying tests

File: src/components/plots/barPlot.ts (hunk 7 above)
Comment (aicioara) [inline src/components/plots/barPlot.ts]: ?

File: src/components/plots/rectanglePlot.ts (hunk 12 above)
Comment (aicioara) [inline src/components/plots/rectanglePlot.ts]: Talked to @rcchen before starting this and he agreed, there was nothing he could think of where x and x2 / y and y2 would use different scales

File: src/components/plots/areaPlot.ts (hunk 4 above)
Comment (bluong) [inline src/components/plots/areaPlot.ts]: The above I believe should already be declared on `Plots.Line` and thus this should be unnecessary?

File: src/components/plots/barPlot.ts (hunk 6 above)
Comment (bluong) [inline src/components/plots/barPlot.ts]: The above I believe should already be declared on XYPlot and thus this should be unnecessary?

File: src/components/plots/xyPlot.ts (hunk 17 above)
Comment (jtlan) [inline src/components/plots/xyPlot.ts]: Adding the callback needs to happen in the property-setter. If it's not happening in any of the other lines in this method, we should do it here.

File: src/components/plots/stackedAreaPlot.ts (hunk 15 above)
Comment (bluong) [inline src/components/plots/stackedAreaPlot.ts]: There needs to be a
```typescript
public x(x? X | Accessor<X>, xScale: QuantitativeScale<X>): Plots.StackedArea
```
above this line

File: src/components/plots/xyPlot.ts (hunk 17 above)
Comment (bluong) [inline src/components/plots/xyPlot.ts]: If we want to leave this TODO, assign a ticket

File: src/components/plots/xyPlot.ts (hunk 19 above)
Comment (bluong) [inline src/components/plots/xyPlot.ts]: this should work i think
```typescript
xBinding != null && xBinding.accessor != null && yBinding != null && yBinding != null;
```

File: src/components/plots/xyPlot.ts (hunk 17 above)
Comment (aicioara) [inline src/components/plots/xyPlot.ts]: It's fine them, will remove the todo

File: src/components/plots/barPlot.ts (hunk 10 above)
Comment (bluong) [inline src/components/plots/barPlot.ts]: Ok understood.  To avoid useless code flow, we should guard with
```typescript
if (!this._isVertical) {
```
right?

File: src/components/plots/areaPlot.ts (hunk 2 above)
Comment (bluong) [inline src/components/plots/areaPlot.ts]: Oh I see.  In this specific circumstance this should be ok.  But in the other scenarios there are now incorrect castings

File: src/components/plots/barPlot.ts (hunk 8 above)
Comment (bluong) [inline src/components/plots/barPlot.ts]: The casting here is not correct

File: src/components/plots/areaPlot.ts (hunk 4 above)
Comment (aicioara) [inline src/components/plots/areaPlot.ts]: Disagree, it aids readability

File: src/components/plots/barPlot.ts (hunk 6 above)
Comment (aicioara) [inline src/components/plots/barPlot.ts]: same

File: src/components/plots/barPlot.ts (hunk 7 above)
Comment (aicioara) [inline src/components/plots/barPlot.ts]: same

File: src/components/plots/barPlot.ts (hunk 10 above)
Comment (aicioara) [inline src/components/plots/barPlot.ts]: It is done in the `this._updateValueScale()`

File: src/components/plots/barPlot.ts [hunk 20]
```diff
@@ -38,6 +38,42 @@ export module Plots {
...
+        return super.x();
+      }
+
+      if (xScale == null) {
+        super.x(<number | Accessor<number>>x);
+      } else {
+        super.x(< X | Accessor<X>>x, xScale);
+      }
+
+      this._updateValueScale();
```
Comment (bluong) [inline src/components/plots/barPlot.ts]: Should there be a guard with `this._isVertical` so we don't have to do needless calculations?

File: src/components/plots/barPlot.ts [hunk 21]
```diff
@@ -38,6 +38,42 @@ export module Plots {
...
+        return super.y();
+      }
+
+      if (yScale == null) {
+        super.y(<number | Accessor<number>>y);
+      } else {
+        super.y(<Y | Accessor<Y>>y, yScale);
+      }
+
+      this._updateValueScale();
```
Comment (bluong) [inline src/components/plots/barPlot.ts]: Should there be a guard with `this._isVertical` so we don't have to do needless calculations?

File: src/components/plots/barPlot.ts (hunk 20 above)
Comment (aicioara) [inline src/components/plots/barPlot.ts]: It is done inside `_updateValueScale()`

File: src/components/plots/barPlot.ts (hunk 21 above)
Comment (aicioara) [inline src/components/plots/barPlot.ts]: same

**Comment by crmorford** [comment]
Did any of my notes from https://github.com/palantir/plottable/issues/1507 get tested? I didn't see any mention of that in the QE notes above

**Comment by aicioara** [comment]
For convenience, writing them here
> Testing Notes:
> * Numeric x Numeric axes
> * Cat x Numeric scales, Numeric x Cat, Cat x Cat
> * Time x Numeric scales, Time x Cat
> * absence of .x() and/or .y()
> *  .x(), .y() called, but leave out scale argument
> * dataset is empty 
> * .x() and .y() get passed same scale
> * plot is rendered, then .x() is called again with a different scale

**Comment by crmorford** [comment]
i took the scales out of the constructor on the animate_area quicktests, and area plots are upside down 
screen shot 2015-05-28 at 12 53 54 pm

**Comment by crmorford** [comment]
If you forget to leave out the scale argument, there should defintely be an error message now. right now there's none.

it also does this: 
screen shot 2015-05-28 at 1 05 44 pm
which is kinda weird because you can see a little blue dot up in the left corner.

**Comment by aicioara** [comment]
@crmorford Can I see the test data? If the numbers are in pixels, the resulting graph is legit

**Comment by crmorford** [comment]
Test Data changes every time, but they are very small numbers, so that makes sense.

File: src/components/plots/linePlot.ts [hunk 22]
```diff
@@ -144,7 +144,9 @@ export module Plots {
 
     protected _propertyProjectors(): AttributeToProjector {
       var propertyToProjectors = super._propertyProjectors();
-      propertyToProjectors["d"] = this._constructLineProjector(Plot._scaledAccessor(this.x()), Plot._scaledAccessor(this.y()));
+      if (this._projectorsReady()) {
```
Comment (aicioara) [inline src/components/plots/linePlot.ts]: @bluong  Here

File: src/components/plots/areaPlot.ts [hunk 23]
```diff
@@ -32,14 +32,39 @@ export module Plots {
...
+
+      if (yScale == null) {
+        super.y(y);
+      } else {
+        super.y(y, yScale);
+      }
+
+      if (yScale != null) {
+        var y0 = this.y0().accessor;
+        if (y0 != null) {
```
Comment (bluong) [inline src/components/plots/areaPlot.ts]: y0 should never be null right?

File: src/components/plots/areaPlot.ts [hunk 24]
```diff
@@ -127,9 +155,11 @@ export module Plots {
 
     protected _propertyProjectors(): AttributeToProjector {
       var propertyToProjectors = super._propertyProjectors();
-      propertyToProjectors["d"] = this._constructAreaProjector(Plot._scaledAccessor(this.x()),
-                                                               Plot._scaledAccessor(this.y()),
-                                                               Plot._scaledAccessor(this.y0()));
+      if (this._projectorsReady()) {
```
Comment (bluong) [inline src/components/plots/areaPlot.ts]: `_propertyProjectors()` is theoretically called after `x` / `y` is set.  Can you look into when this is called?

File: src/components/plots/linePlot.ts (hunk 22 above)
Comment (bluong) [inline src/components/plots/linePlot.ts]: Check when this function is called and see why this guard is needed

Comment (aicioara) [inline src/components/plots/linePlot.ts]: Fixed

File: src/components/plots/areaPlot.ts (hunk 23 above)
Comment (aicioara) [inline src/components/plots/areaPlot.ts]: Too late, will fix later

**Comment by aicioara** [comment]
Need review for the overlaying test fix.

File: src/components/plots/areaPlot.ts (hunk 23 above)
Comment (bluong) [inline src/components/plots/areaPlot.ts]: Should be a simple fix no?  Just remove the `y0 != null`

File: src/components/plots/barPlot.ts [hunk 25]
```diff
@@ -315,6 +351,9 @@ export module Plots {
     }
 
     private _updateValueScale() {
+      if (!this._projectorsReady()) {
+        return;
+      }
```
Comment (bluong) [inline src/components/plots/barPlot.ts]: Why is this here?  Wouldn't a better catch be to move this one line below and break out if valueScale is null?

Comment (aicioara) [inline src/components/plots/barPlot.ts]: it is called by the constructor, through `baseline()`. Hence guard needed.
I would prefer not to do any code change anymore as that needs QE again.

File: src/components/plots/areaPlot.ts (hunk 23 above)
Comment (aicioara) [inline src/components/plots/areaPlot.ts]: I prefer not to change code anymore, as it would need QE time again.

File: src/components/plots/areaPlot.ts (hunk 24 above)
Comment (aicioara) [inline src/components/plots/areaPlot.ts]: This has been removed at some later point
